### PR TITLE
Testing Category Page

### PR DESCRIPTION
**Pull request by sofiengwin**
## Why

Confirms the category page is functioning properly
## How

 Test that product displayed matches selected category
Test the classes that implement category page

https://www.pivotaltracker.com/story/show/114657453

[finishes #114657453]

### CODE REVIEW

File: spec/controllers/category_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe CategoriesController, type: :controller do
+  let(:categories) { FactoryGirl.create(:category) }
+  describe "GET index" do
+    it "it renders index action" do
```
Comment (makinwab) [inline spec/controllers/category_controller_spec.rb]: **Nitpick - Code Quality**
You don't need to append `it` to the description.
You can have `it "renders based on the index action" do`

Please fix all instances of this issue.

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -1,16 +1,21 @@
 require "rails_helper"
 
 RSpec.describe Category, type: :model do
-  let(:category) { build(:category) }
-
-  context "when initializing product category" do
+  context "when initializing a category" do
+    before :all do
+      @category = create(:category)
+    end
```
Comment (makinwab) [inline spec/models/category_spec.rb]: **Nitpick - Code Quality**
Add a newline after this line.

Comment (sofiengwin) [inline spec/models/category_spec.rb]: Fixed

**Comment by codesword** [comment]
@andela-gogbara Why are there so many merge commit in this PR?

**Pull request merged**